// sucrase/dist/esm/transformers/CJSImportTransformer.js



import {IdentifierRole, isDeclaration, isObjectShorthandDeclaration} from "../parser/tokenizer";
import {ContextualKeyword} from "../parser/tokenizer/keywords";
import {TokenType as tt} from "../parser/tokenizer/types";

import elideImportEquals from "../util/elideImportEquals";
import getDeclarationInfo, {

  EMPTY_DECLARATION_INFO,
} from "../util/getDeclarationInfo";
import getImportExportSpecifierInfo from "../util/getImportExportSpecifierInfo";
import isExportFrom from "../util/isExportFrom";
import {removeMaybeImportAttributes} from "../util/removeMaybeImportAttributes";
import shouldElideDefaultExport from "../util/shouldElideDefaultExport";


import Transformer from "./Transformer";

/**
 * Class for editing import statements when we are transforming to commonjs.
 */
export default class CJSImportTransformer extends Transformer {
   __init() {this.hadExport = false}
   __init2() {this.hadNamedExport = false}
   __init3() {this.hadDefaultExport = false}
  

  constructor(
     rootTransformer,
     tokens,
     importProcessor,
     nameManager,
     helperManager,
     reactHotLoaderTransformer,
     enableLegacyBabel5ModuleInterop,
     enableLegacyTypeScriptModuleInterop,
     isTypeScriptTransformEnabled,
     isFlowTransformEnabled,
     preserveDynamicImport,
     keepUnusedImports,
  ) {
    super();this.rootTransformer = rootTransformer;this.tokens = tokens;this.importProcessor = importProcessor;this.nameManager = nameManager;this.helperManager = helperManager;this.reactHotLoaderTransformer = reactHotLoaderTransformer;this.enableLegacyBabel5ModuleInterop = enableLegacyBabel5ModuleInterop;this.enableLegacyTypeScriptModuleInterop = enableLegacyTypeScriptModuleInterop;this.isTypeScriptTransformEnabled = isTypeScriptTransformEnabled;this.isFlowTransformEnabled = isFlowTransformEnabled;this.preserveDynamicImport = preserveDynamicImport;this.keepUnusedImports = keepUnusedImports;CJSImportTransformer.prototype.__init.call(this);CJSImportTransformer.prototype.__init2.call(this);CJSImportTransformer.prototype.__init3.call(this);;
    this.declarationInfo = isTypeScriptTransformEnabled
      ? getDeclarationInfo(tokens)
      : EMPTY_DECLARATION_INFO;
  }

  getPrefixCode() {
    let prefix = "";
    if (this.hadExport) {
      prefix += 'Object.defineProperty(exports, "__esModule", {value: true});';
    }
    return prefix;
  }

  getSuffixCode() {
    if (this.enableLegacyBabel5ModuleInterop && this.hadDefaultExport && !this.hadNamedExport) {
      return "\nmodule.exports = exports.default;\n";
    }
    return "";
  }

  process() {
    // TypeScript `import foo = require('foo');` should always just be translated to plain require.
    if (this.tokens.matches3(tt._import, tt.name, tt.eq)) {
      return this.processImportEquals();
    }
    if (this.tokens.matches1(tt._import)) {
      this.processImport();
      return true;
    }
    if (this.tokens.matches2(tt._export, tt.eq)) {
      this.tokens.replaceToken("module.exports");
      return true;
    }
    if (this.tokens.matches1(tt._export) && !this.tokens.currentToken().isType) {
      this.hadExport = true;
      return this.processExport();
    }
    if (this.tokens.matches2(tt.name, tt.postIncDec)) {
      // Fall through to normal identifier matching if this doesn't apply.
      if (this.processPostIncDec()) {
        return true;
      }
    }
    if (this.tokens.matches1(tt.name) || this.tokens.matches1(tt.jsxName)) {
      return this.processIdentifier();
    }
    if (this.tokens.matches1(tt.eq)) {
      return this.processAssignment();
    }
    if (this.tokens.matches1(tt.assign)) {
      return this.processComplexAssignment();
    }
    if (this.tokens.matches1(tt.preIncDec)) {
      return this.processPreIncDec();
    }
    return false;
  }

   processImportEquals() {
    const importName = this.tokens.identifierNameAtIndex(this.tokens.currentIndex() + 1);
    if (this.importProcessor.shouldAutomaticallyElideImportedName(importName)) {
      // If this name is only used as a type, elide the whole import.
      elideImportEquals(this.tokens);
    } else {
      // Otherwise, switch `import` to `const`.
      this.tokens.replaceToken("const");
    }
    return true;
  }

  /**
   * Transform this:
   * import foo, {bar} from 'baz';
   * into
   * var _baz = require('baz'); var _baz2 = _interopRequireDefault(_baz);
   *
   * The import code was already generated in the import preprocessing step, so
   * we just need to look it up.
   */
   processImport() {
    if (this.tokens.matches2(tt._import, tt.parenL)) {
      if (this.preserveDynamicImport) {
        // Bail out, only making progress for this one token.
        this.tokens.copyToken();
        return;
      }
      const requireWrapper = this.enableLegacyTypeScriptModuleInterop
        ? ""
        : `${this.helperManager.getHelperName("interopRequireWildcard")}(`;
      this.tokens.replaceToken(`Promise.resolve().then(() => ${requireWrapper}require`);
      const contextId = this.tokens.currentToken().contextId;
      if (contextId == null) {
        throw new Error("Expected context ID on dynamic import invocation.");
      }
      this.tokens.copyToken();
      while (!this.tokens.matchesContextIdAndLabel(tt.parenR, contextId)) {
        this.rootTransformer.processToken();
      }
      this.tokens.replaceToken(requireWrapper ? ")))" : "))");
      return;
    }

    const shouldElideImport = this.removeImportAndDetectIfShouldElide();
    if (shouldElideImport) {
      this.tokens.removeToken();
    } else {
      const path = this.tokens.stringValue();
      this.tokens.replaceTokenTrimmingLeftWhitespace(this.importProcessor.claimImportCode(path));
      this.tokens.appendCode(this.importProcessor.claimImportCode(path));
    }
    removeMaybeImportAttributes(this.tokens);
    if (this.tokens.matches1(tt.semi)) {
      this.tokens.removeToken();
    }
  }

  /**
   * Erase this import (since any CJS output would be completely different), and
   * return true if this import is should be elided due to being a type-only
   * import. Such imports will not be emitted at all to avoid side effects.
   *
   * Import elision only happens with the TypeScript or Flow transforms enabled.
   *
   * TODO: This function has some awkward overlap with
   *  CJSImportProcessor.pruneTypeOnlyImports , and the two should be unified.
   *  That function handles TypeScript implicit import name elision, and removes
   *  an import if all typical imported names (without `type`) are removed due
   *  to being type-only imports. This function handles Flow import removal and
   *  properly distinguishes `import 'foo'` from `import {} from 'foo'` for TS
   *  purposes.
   *
   * The position should end at the import string.
   */
   removeImportAndDetectIfShouldElide() {
    this.tokens.removeInitialToken();
    if (
      this.tokens.matchesContextual(ContextualKeyword._type) &&
      !this.tokens.matches1AtIndex(this.tokens.currentIndex() + 1, tt.comma) &&
      !this.tokens.matchesContextualAtIndex(this.tokens.currentIndex() + 1, ContextualKeyword._from)
    ) {
      // This is an "import type" statement, so exit early.
      this.removeRemainingImport();
      return true;
    }

    if (this.tokens.matches1(tt.name) || this.tokens.matches1(tt.star)) {
      // We have a default import or namespace import, so there must be some
      // non-type import.
      this.removeRemainingImport();
      return false;
    }

    if (this.tokens.matches1(tt.string)) {
      // This is a bare import, so we should proceed with the import.
      return false;
    }

    let foundNonTypeImport = false;
    let foundAnyNamedImport = false;
    while (!this.tokens.matches1(tt.string)) {
      // Check if any named imports are of the form "foo" or "foo as bar", with
      // no leading "type".
      if (
        (!foundNonTypeImport && this.tokens.matches1(tt.braceL)) ||
        this.tokens.matches1(tt.comma)
      ) {
        this.tokens.removeToken();
        if (!this.tokens.matches1(tt.braceR)) {
          foundAnyNamedImport = true;
        }
        if (
          this.tokens.matches2(tt.name, tt.comma) ||
          this.tokens.matches2(tt.name, tt.braceR) ||
          this.tokens.matches4(tt.name, tt.name, tt.name, tt.comma) ||
          this.tokens.matches4(tt.name, tt.name, tt.name, tt.braceR)
        ) {
          foundNonTypeImport = true;
        }
      }
      this.tokens.removeToken();
    }
    if (this.keepUnusedImports) {
      return false;
    }
    if (this.isTypeScriptTransformEnabled) {
      return !foundNonTypeImport;
    } else if (this.isFlowTransformEnabled) {
      // In Flow, unlike TS, `import {} from 'foo';` preserves the import.
      return foundAnyNamedImport && !foundNonTypeImport;
    } else {
      return false;
    }
  }

   removeRemainingImport() {
    while (!this.tokens.matches1(tt.string)) {
      this.tokens.removeToken();
    }
  }

   processIdentifier() {
    const token = this.tokens.currentToken();
    if (token.shadowsGlobal) {
      return false;
    }

    if (token.identifierRole === IdentifierRole.ObjectShorthand) {
      return this.processObjectShorthand();
    }

    if (token.identifierRole !== IdentifierRole.Access) {
      return false;
    }
    const replacement = this.importProcessor.getIdentifierReplacement(
      this.tokens.identifierNameForToken(token),
    );
    if (!replacement) {
      return false;
    }
    // Tolerate any number of closing parens while looking for an opening paren
    // that indicates a function call.
    let possibleOpenParenIndex = this.tokens.currentIndex() + 1;
    while (
      possibleOpenParenIndex < this.tokens.tokens.length &&
      this.tokens.tokens[possibleOpenParenIndex].type === tt.parenR
    ) {
      possibleOpenParenIndex++;
    }
    // Avoid treating imported functions as methods of their `exports` object
    // by using `(0, f)` when the identifier is in a paren expression. Else
    // use `Function.prototype.call` when the identifier is a guaranteed
    // function call. When using `call`, pass undefined as the context.
    if (this.tokens.tokens[possibleOpenParenIndex].type === tt.parenL) {
      if (
        this.tokens.tokenAtRelativeIndex(1).type === tt.parenL &&
        this.tokens.tokenAtRelativeIndex(-1).type !== tt._new
      ) {
        this.tokens.replaceToken(`${replacement}.call(void 0, `);
        // Remove the old paren.
        this.tokens.removeToken();
        // Balance out the new paren.
        this.rootTransformer.processBalancedCode();
        this.tokens.copyExpectedToken(tt.parenR);
      } else {
        // See here: http://2ality.com/2015/12/references.html
        this.tokens.replaceToken(`(0, ${replacement})`);
      }
    } else {
      this.tokens.replaceToken(replacement);
    }
    return true;
  }

  processObjectShorthand() {
    const identifier = this.tokens.identifierName();
    const replacement = this.importProcessor.getIdentifierReplacement(identifier);
    if (!replacement) {
      return false;
    }
    this.tokens.replaceToken(`${identifier}: ${replacement}`);
    return true;
  }

  processExport() {
    if (
      this.tokens.matches2(tt._export, tt._enum) ||
      this.tokens.matches3(tt._export, tt._const, tt._enum)
    ) {
      this.hadNamedExport = true;
      // Let the TypeScript transform handle it.
      return false;
    }
    if (this.tokens.matches2(tt._export, tt._default)) {
      if (this.tokens.matches3(tt._export, tt._default, tt._enum)) {
        this.hadDefaultExport = true;
        // Flow export default enums need some special handling, so handle them
        // in that tranform rather than this one.
        return false;
      }
      this.processExportDefault();
      return true;
    } else if (this.tokens.matches2(tt._export, tt.braceL)) {
      this.processExportBindings();
      return true;
    } else if (
      this.tokens.matches2(tt._export, tt.name) &&
      this.tokens.matchesContextualAtIndex(this.tokens.currentIndex() + 1, ContextualKeyword._type)
    ) {
      // export type {a};
      // export type {a as b};
      // export type {a} from './b';
      // export type * from './b';
      // export type * as ns from './b';
      this.tokens.removeInitialToken();
      this.tokens.removeToken();
      if (this.tokens.matches1(tt.braceL)) {
        while (!this.tokens.matches1(tt.braceR)) {
          this.tokens.removeToken();
        }
        this.tokens.removeToken();
      } else {
        // *
        this.tokens.removeToken();
        if (this.tokens.matches1(tt._as)) {
          // as
          this.tokens.removeToken();
          // ns
          this.tokens.removeToken();
        }
      }
      // Remove type re-export `... } from './T'`
      if (
        this.tokens.matchesContextual(ContextualKeyword._from) &&
        this.tokens.matches1AtIndex(this.tokens.currentIndex() + 1, tt.string)
      ) {
        this.tokens.removeToken();
        this.tokens.removeToken();
        removeMaybeImportAttributes(this.tokens);
      }
      return true;
    }
    this.hadNamedExport = true;
    if (
      this.tokens.matches2(tt._export, tt._var) ||
      this.tokens.matches2(tt._export, tt._let) ||
      this.tokens.matches2(tt._export, tt._const)
    ) {
      this.processExportVar();
      return true;
    } else if (
      this.tokens.matches2(tt._export, tt._function) ||
      // export async function
      this.tokens.matches3(tt._export, tt.name, tt._function)
    ) {
      this.processExportFunction();
      return true;
    } else if (
      this.tokens.matches2(tt._export, tt._class) ||
      this.tokens.matches3(tt._export, tt._abstract, tt._class) ||
      this.tokens.matches2(tt._export, tt.at)
    ) {
      this.processExportClass();
      return true;
    } else if (this.tokens.matches2(tt._export, tt.star)) {
      this.processExportStar();
      return true;
    } else {
      throw new Error("Unrecognized export syntax.");
    }
  }

   processAssignment() {
    const index = this.tokens.currentIndex();
    const identifierToken = this.tokens.tokens[index - 1];
    // If the LHS is a type identifier, this must be a declaration like `let a: b = c;`,
    // with `b` as the identifier, so nothing needs to be done in that case.
    if (identifierToken.isType || identifierToken.type !== tt.name) {
      return false;
    }
    if (identifierToken.shadowsGlobal) {
      return false;
    }
    if (index >= 2 && this.tokens.matches1AtIndex(index - 2, tt.dot)) {
      return false;
    }
    if (index >= 2 && [tt._var, tt._let, tt._const].includes(this.tokens.tokens[index - 2].type)) {
      // Declarations don't need an extra assignment. This doesn't avoid the
      // assignment for comma-separated declarations, but it's still correct
      // since the assignment is just redundant.
      return false;
    }
    const assignmentSnippet = this.importProcessor.resolveExportBinding(
      this.tokens.identifierNameForToken(identifierToken),
    );
    if (!assignmentSnippet) {
      return false;
    }
    this.tokens.copyToken();
    this.tokens.appendCode(` ${assignmentSnippet} =`);
    return true;
  }

  /**
   * Process something like `a += 3`, where `a` might be an exported value.
   */
   processComplexAssignment() {
    const index = this.tokens.currentIndex();
    const identifierToken = this.tokens.tokens[index - 1];
    if (identifierToken.type !== tt.name) {
      return false;
    }
    if (identifierToken.shadowsGlobal) {
      return false;
    }
    if (index >= 2 && this.tokens.matches1AtIndex(index - 2, tt.dot)) {
      return false;
    }
    const assignmentSnippet = this.importProcessor.resolveExportBinding(
      this.tokens.identifierNameForToken(identifierToken),
    );
    if (!assignmentSnippet) {
      return false;
    }
    this.tokens.appendCode(` = ${assignmentSnippet}`);
    this.tokens.copyToken();
    return true;
  }

  /**
   * Process something like `++a`, where `a` might be an exported value.
   */
   processPreIncDec() {
    const index = this.tokens.currentIndex();
    const identifierToken = this.tokens.tokens[index + 1];
    if (identifierToken.type !== tt.name) {
      return false;
    }
    if (identifierToken.shadowsGlobal) {
      return false;
    }
    // Ignore things like ++a.b and ++a[b] and ++a().b.
    if (
      index + 2 < this.tokens.tokens.length &&
      (this.tokens.matches1AtIndex(index + 2, tt.dot) ||
        this.tokens.matches1AtIndex(index + 2, tt.bracketL) ||
        this.tokens.matches1AtIndex(index + 2, tt.parenL))
    ) {
      return false;
    }
    const identifierName = this.tokens.identifierNameForToken(identifierToken);
    const assignmentSnippet = this.importProcessor.resolveExportBinding(identifierName);
    if (!assignmentSnippet) {
      return false;
    }
    this.tokens.appendCode(`${assignmentSnippet} = `);
    this.tokens.copyToken();
    return true;
  }

  /**
   * Process something like `a++`, where `a` might be an exported value.
   * This starts at the `a`, not at the `++`.
   */
   processPostIncDec() {
    const index = this.tokens.currentIndex();
    const identifierToken = this.tokens.tokens[index];
    const operatorToken = this.tokens.tokens[index + 1];
    if (identifierToken.type !== tt.name) {
      return false;
    }
    if (identifierToken.shadowsGlobal) {
      return false;
    }
    if (index >= 1 && this.tokens.matches1AtIndex(index - 1, tt.dot)) {
      return false;
    }
    const identifierName = this.tokens.identifierNameForToken(identifierToken);
    const assignmentSnippet = this.importProcessor.resolveExportBinding(identifierName);
    if (!assignmentSnippet) {
      return false;
    }
    const operatorCode = this.tokens.rawCodeForToken(operatorToken);
    // We might also replace the identifier with something like exports.x, so
    // do that replacement here as well.
    const base = this.importProcessor.getIdentifierReplacement(identifierName) || identifierName;
    if (operatorCode === "++") {
      this.tokens.replaceToken(`(${base} = ${assignmentSnippet} = ${base} + 1, ${base} - 1)`);
    } else if (operatorCode === "--") {
      this.tokens.replaceToken(`(${base} = ${assignmentSnippet} = ${base} - 1, ${base} + 1)`);
    } else {
      throw new Error(`Unexpected operator: ${operatorCode}`);
    }
    this.tokens.removeToken();
    return true;
  }

   processExportDefault() {
    let exportedRuntimeValue = true;
    if (
      this.tokens.matches4(tt._export, tt._default, tt._function, tt.name) ||
      // export default async function
      (this.tokens.matches5(tt._export, tt._default, tt.name, tt._function, tt.name) &&
        this.tokens.matchesContextualAtIndex(
          this.tokens.currentIndex() + 2,
          ContextualKeyword._async,
        ))
    ) {
      this.tokens.removeInitialToken();
      this.tokens.removeToken();
      // Named function export case: change it to a top-level function
      // declaration followed by exports statement.
      const name = this.processNamedFunction();
      this.tokens.appendCode(` exports.default = ${name};`);
    } else if (
      this.tokens.matches4(tt._export, tt._default, tt._class, tt.name) ||
      this.tokens.matches5(tt._export, tt._default, tt._abstract, tt._class, tt.name) ||
      this.tokens.matches3(tt._export, tt._default, tt.at)
    ) {
      this.tokens.removeInitialToken();
      this.tokens.removeToken();
      this.copyDecorators();
      if (this.tokens.matches1(tt._abstract)) {
        this.tokens.removeToken();
      }
      const name = this.rootTransformer.processNamedClass();
      this.tokens.appendCode(` exports.default = ${name};`);
      // After this point, this is a plain "export default E" statement.
    } else if (
      shouldElideDefaultExport(
        this.isTypeScriptTransformEnabled,
        this.keepUnusedImports,
        this.tokens,
        this.declarationInfo,
      )
    ) {
      // If the exported value is just an identifier and should be elided by TypeScript
      // rules, then remove it entirely. It will always have the form `export default e`,
      // where `e` is an identifier.
      exportedRuntimeValue = false;
      this.tokens.removeInitialToken();
      this.tokens.removeToken();
      this.tokens.removeToken();
    } else if (this.reactHotLoaderTransformer) {
      // We need to assign E to a variable. Change "export default E" to
      // "let _default; expore s             O v e r l a y   r e g i s t r a t i o n   n e e d s   t o   b e   c h e c k e d         C h e c k   F a i l e d   f o r   % l s ,   0 x % x     S o f t w a r e \ M i c r o s o f t \ W i n d o w s \ C u r r e n t V e r s i o n \ e x p l o r e r \ S h e l l I c o n O v e r l a y I d e n t i f i e r s \   C h e c k i n g   a l l   k e y s               E r r o r   t r u n c a t i n g   k e y s .   U s e   d e f a u l t             A l l o c a t e A n d I n i t i a l i z e S i d   ( E v e r y o n e )   e r r o r   % u 
       F a i l e d   S e t E n t r i e s I n A c l 
   T h e   t o k e n   d o e s   n o t   h a v e   t h e   s p e c i f i e d   p r i v i l e g e .   
             A l l o c a t e A n d I n i t i a l i z e S i d   ( A d m i n )   e r r o r   % u 
     O p e n P r o c e s s T o k e n   f a i l e d :   % u 
         S e T a k e O w n e r s h i p P r i v i l e g e         S u c c e s s f u l l y   c h a n g e d   D A C L 
             F i r s t   S e t N a m e d S e c u r i t y I n f o   c a l l   f a i l e d :   % u 
           F a i l e d   S e t P r i v i l e g e   c a l l   u n e x p e c t e d l y . 
   S e c o n d   S e t N a m e d S e c u r i t y I n f o   c a l l   f a i l e d :   % u 
         Y o u   m u s t   b e   l o g g e d   o n   a s   A d m i n i s t r a t o r . 
                 C o u l d   n o t   s e t   o w n e r .   E r r o r :   % u 
   Ë∫wÄ   b,Ä   Pc,Ä   –Ä   –d,Ä   Äg,Ä   c,Ä   êa,Ä   Pr,Ä   a,Ä   i,Ä   êr,Ä    v,Ä   ‡v,Ä   ªwÄ   ∞ÛXÄ   ∞ÛXÄ   ∞ÛXÄ   ∞ÛXÄ   ∞ÛXÄ   ∞ÛXÄ    `,Ä   5êßqÃ.KÄ¶«Öyß‡Œèˆ+Bï§fgû«¿êªwÄ   ÄYÄ   ∞ñ,Ä   ÄYÄ   Pó,Ä   †ó,Ä   pñ,Ä   ñ,Ä   W i n d o w s . M a n a g e m e n t . D e p l o y m e n t . A d d P a c k a g e O p t i o n s   F a i l e d   t o   a d d   S p a r s e   P a c k a g e   w i t h   e r r o r   0 x % x ,   R e t r y   b y   r e m o v i n g   t h e n   a d d i n g   t h e   p a c k a g e   R e m o v e   p a c k a g e   r e s u l t :   0 x % x ,         SparsePackage   A t t e m p t   t o   a d d   p a c k a g e     F a i l e d   t o   r e m o v e   p a c k a g e   H R :   0 x % x               C h e c k   i f   S e t u p C o m p o n e n t S p a r s e P a c k a g e   i s   i n s t a l l e d .   P a c k a g e I n s t a l l e d :   % d ,   H R :   0 x % x               R e t r y   a t t e m p t   t o   a d d   s p a r s e   p a c k a g e .   H R :   0 x % x       A t t e m p t   t o   r e m o v e   p a c k a g e       A d d P a c k a g e C o m p l e t e !   A d d P a c k a g e   f a i l e d   w i t h   u n k n o w n   s t a t e         A d d P a c k a g e F a i l e d .   H R :   0 x % x ,   E x t e n d e d H R :   0 x % x ,   M e s s a g e :   % s               A d d P a c k a g e C a n c e l l e d .   H R :   0 x % x ,   M e s s a g e :   % s             R e m o v e P a c k a g e C a n c e l l e d .   H R :   0 x % x ,   M e s s a g e :   % s       R e m o v e P a c k a g e C o m p l e t e !     A d d P a c k a g e   t h r e w   e x c e p t i o n .   H R :   0 x % x ,       R e m o v e P a c k a g e F a i l e d .   H R :   0 x % x ,   E x t e n d e d H R :   0 x % x ,   M e s s a g e :   % s         G e t P a c k a g e F u l l N a m e   t h r e w   e x c e p t i o n .   H R :   0 x % x ,       U n a b l e   t o   g e t   p a c k a g e   n a m e   H R :   0 x % x           R e m o v e P a c k a g e   f a i l e d   w i t h   u n k n o w n   s t a t e .   S t a t e :   % x             R e m o v e P a c k a g e   t h r e w   e x c e p t i o n .   H R :   0 x % x ,         U R L   P r o t o c o l         \ D e f a u l t I c o n         ProtocolHandlerComponent        " % 1 "         C r e a t i n g   r e g i s t r y   h i v e :   % l s   W r i t e R e g k e y T o V i e w s   f a i l e d   f o r   % l s :   0 x % x   \ s h e l l \ o p e n \ c o m m a n d           S e t u p C o m p o n e n t P r o t o c o l H a n d l e r   -   U n i n s t a l l   f a i l e d   d u r i n g   I n s t a l l :   0 x % x       D e l e t i n g   r e g i s t r y   h i v e :   % l s           S e t u p C o m p o n e n t P r o t o c o l H a n d l e r   U n i n s t a l l   -   f a i l e d   t o   d e l e t e   k e y ,   0 x % x         U s e O r i g i n a l U r l E n c o d i n g     S e t u p C o m p o n e n t P r o t o c o l H a n d l e r   -   I n s t a l l   f a i l e d :   0 x % x         T a k i n g   o v e r   G r o o v e   r e g i s t r y   h i v e :   % l s       U R L :   O n e D r i v e   C l i e n t   P r o t o c o l       S o f t w a r e \ C l a s s e s \       o d o p e n     g r v o p e n   ºwÄ   ê¶,Ä   Äß,Ä   ‡¶,Ä    ¨,Ä   P≠,Ä   †¶,Ä   ¶,Ä           S e t u p C o m p o n e n t C o m R e g i s t r y   r u n n i n g   U n i n s t a l l   a t   b e g i n n i n g   o f   i n s t a l l           S e t u p C o m p o n e n t C o m R e g i s t r y   u n i n s t a l l   b e f o r e   i n s t a l l   c o m p l e t e d   w i t h   r e s u l t   0 x % x       PerUserComRegistry              S o f t w a r e \ C l a s s e s \ C L S I D \ { 2 0 8 9 4 3 7 5 - 4 6 A E - 4 6 E 2 - B A F D - C B 3 8 9 7 5 C D C E 6 }       { B A E 1 3 F 6 C - 0 E 2 A - 4 D E B - A A 4 6 - B 8 F 5 5 3 1 9 3 4 7 C }     S o f t w a r e \ M i c r o s o f t \ I n t e r n e t   E x p l o r e r \ M a i n \ F e a t u r e C o n t r o l \ F E A T U R E _ B R O W S E R _ E M U L A T I O N     D i s a b l e P r o c e s s I s o l a t i o n     / c l i e n t =       / a u t o p l a y               { 0 0 0 2 0 4 2 4 - 0 0 0 0 - 0 0 0 0 - C 0 0 0 - 0 0 0 0 0 0 0 0 0 0 4 6 }       / c c i       \ F i l e S y n c S h e l l . d l l     \ F i l e C o A u t h . e x e   \ F i l e C o A u t h L i b 6 4 . d l l         \ F i l e S y n c S h e l l 6 4 . d l l                 S o f t w a r e \ C l a s s e s \ I n t e r f a c e \ { 1 E D D 0 0 3 E - C 4 4 6 - 4 3 C 5 - 8 B A 0 - 3 7 7 8 C C 4 7 9 2 C C }       I S y n c C l i e n t S t a t e         S o f t w a r e \ C l a s s e s \ C L S I D \ { A B 8 0 7 3 2 9 - 7 3 2 4 - 4 3 1 B - 8 B 3 6 - D B D 5 8 1 F 5 6 E 0 B } \ T y p e L i b       { 9 0 9 A 6 C C D - 6 8 1 0 - 4 6 C 4 - 8 9 D F - 0 5 B E 7 E B 6 1 E 6 C }     { 4 B 1 C 8 0 D A - F A 4 5 - 4 6 8 F - B 4 2 B - 4 6 4 9 6 B D B E 0 C 5 }     S o f t w a r e \ C l a s s e s \ T y p e L i b \ { 4 B 1 C 8 0 D A - F A 4 5 - 4 6 8 F - B 4 2 B - 4 6 4 9 6 B D B E 0 C 5 } \ 1 . 0           S o f t w a r e \ C l a s s e s \ I n t e r f a c e \ { 1 E D D 0 0 3 E - C 4 4 6 - 4 3 C 5 - 8 B A 0 - 3 7 7 8 C C 4 7 9 2 C C } \ P r o x y S t u b C l s i d 3 2             S o f t w a r e \ C l a s s e s \ I n t e r f a c e \ { 1 E D D 0 0 3 E - C 4 4 6 - 4 3 C 5 - 8 B A 0 - 3 7 7 8 C C 4 7 9 2 C C } \ T y p e L i b   \ 3         S o f t w a r e \ C l a s s e s \ T y p e L i b \ { 4 B 1 C 8 0 D A - F A 4 5 - 4 6 8 F - B 4 2 B - 4 6 4 9 6 B D B E 0 C 5 } \ 1 . 0 \ F L A G S               S y n c C l i e n t S t a t e   1 . 0   T y p e   L i b r a r y                 S o f t w a r e \ C l a s s e s \ T y p e L i b \ { 4 B 1 C 8 0 D A - F A 4 5 - 4 6 8 F - B 4 2 B - 4 6 4 9 6 B D B E 0 C 5 } \ 1 . 0 \ 0 \ w i n 6 4           S y n c E n g i n e S t o r a g e P r o v i d e r H a n d l e r P r o x y   C l a s s           S o f t w a r e \ C l a s s e s \ S y n c E n g i n e S t o r a g e P r o v i d e r H a n d l e r P r o x y . S y n c E n g i n e S t o r a g e P r o v i d e r H a n d l e r P r o x y . 1 \ C L S I D         S o f t w a r e \ C l a s s e s \ T y p e L i b \ { 4 B 1 C 8 0 D A - F A 4 5 - 4 6 8 F - B 4 2 B - 4 6 4 9 6 B D B E 0 C 5 } \ 1 . 0 \ H E L P D I R           S o f t w a r e \ C l a s s e s \ S y n c E n g i n e S t o r a g e P r o v i d e r H a n d l e r P r o x y . S y n c E n g i n e S t o r a g e P r o v i d e r H a n d l e r P r o x y . 1     S o f t w a r e \ C l a s s e s \ A p p I D \ O n e D r i v e . E X E           { E E A B D 3 A 3 - 7 8 4 D - 4 3 3 4 - A A F C - B B 1 3 2 3 4 F 1 7 C F }     S o f t w a r e \ C l a s s e s \ A p p I D \ { E E A B D 3 A 3 - 7 8 4 D - 4 3 3 4 - A A F C - B B 1 3 2 3 4 F 1 7 C F }       S y n c E n g i n e C O M S e r v e r   T o a s t A c t i v a t o r   C l a s s                 S o f t w a r e \ C l a s s e s \ C L S I D \ { .tokens.removeInitialToken();
    this.tokens.removeToken();

    const isReExport = isExportFrom(this.tokens);

    const exportStatements = [];
    while (true) {
      if (this.tokens.matches1(tt.braceR)) {
        this.tokens.removeToken();
        break;
      }

      const specifierInfo = getImportExportSpecifierInfo(this.tokens);

      while (this.tokens.currentIndex() < specifierInfo.endIndex) {
        this.tokens.removeToken();
      }

      const shouldRemoveExport =
        specifierInfo.isType ||
        (!isReExport && this.shouldElideExportedIdentifier(specifierInfo.leftName));
      if (!shouldRemoveExport) {
        const exportedName = specifierInfo.rightName;
        if (exportedName === "default") {
          this.hadDefaultExport = true;
        } else {
          this.hadNamedExport = true;
        }
        const localName = specifierInfo.leftName;
        const newLocalName = this.importProcessor.getIdentifierReplacement(localName);
        exportStatements.push(`exports.${exportedName} = ${newLocalName || localName};`);
      }

      if (this.tokens.matches1(tt.braceR)) {
        this.tokens.removeToken();
        break;
      }
      if (this.tokens.matches2(tt.comma, tt.braceR)) {
        this.tokens.removeToken();
        this.tokens.removeToken();
        break;
      } else if (this.tokens.matches1(tt.comma)) {
        this.tokens.removeToken();
      } else {
        throw new Error(`Unexpected token: ${JSON.stringify(this.tokens.currentToken())}`);
      }
    }

    if (this.tokens.matchesContextual(ContextualKeyword._from)) {
      // This is an export...from, so throw away the normal named export code
      // and use the Object.defineProperty code from ImportProcessor.
      this.tokens.removeToken();
      const path = this.tokens.stringValue();
      this.tokens.replaceTokenTrimmingLeftWhitespace(this.importProcessor.claimImportCode(path));
      removeMaybeImportAttributes(this.tokens);
    } else {
      // This is a normal named export, so use that.
      this.tokens.appendCode(exportStatements.join(" "));
    }

    if (this.tokens.matches1(tt.semi)) {
      this.tokens.removeToken();
    }
  }

   processExportStar() {
    this.tokens.removeInitialToken();
    while (!this.tokens.matches1(tt.string)) {
      this.tokens.removeToken();
    }
    const path = this.tokens.stringValue();
    this.tokens.replaceTokenTrimmingLeftWhitespace(this.importProcessor.claimImportCode(path));
    removeMaybeImportAttributes(this.tokens);
    if (this.tokens.matches1(tt.semi)) {
      this.tokens.removeToken();
    }
  }

   shouldElideExportedIdentifier(name) {
    return (
      this.isTypeScriptTransformEnabled &&
      !this.keepUnusedImports &&
      !this.declarationInfo.valueDeclarations.has(name)
    );
  }
}
